src/versioninfo.jl

```diff
@@ -0,0 +1,33 @@
+"""
+Print some useful information about the current RCall.jl setup.
+If `all` is false, only RCall.jl information is printed, otherwise
+information about your Julia and R setup is printed as well.
+"""
+function versioninfo(; all=true)
+    if all
+        println("RCall.jl information:")
+    end
+    version = pkgversion(RCall)
+    prefix = all ? "\t" : ""
+    println("$(prefix)RCall.jl version $version")
+    println("$(prefix)R location configured with Preferences.jl: $Rhome_set_as_preference")
+    println("$(prefix)RCall.jl managed R using Conda.jl: $conda_provided_r")
+    println("$(prefix)Rhome: $Rhome")
+    println("$(prefix)libR: $libR")
+    if all
+        println()
+        println("Julia information:")
+        buffer = IOBuffer()
+        Main.versioninfo(buffer)
```

Comment:
package code should not generally call out to `Main` -- `Main` doesn't exist in some sense from the package's perspective and it can't be listed as a dependency. `versioninfo` is actually defined in `InteractiveUtils`, so if we were doing to do this, I would define a stub

```julia
function versioninfo end
```

in the main package and then define an appropriate package extension that defines the full method when `InteractiveUtils` will be loaded, which is generally the case when the user is running stuff interactively, but not the case if a user is running e.g. 

```sh
julia -e'using RCall; RCall.versioninfo()'
```

Reply:
Okay, I can move it to a package extension, but I think it's still useful to keep it, and have one function which shows everything also for people to check the current state themselves. The idea is to show where R has come from and how it's managed by RCall.jl. This could be extended later to try and detect some of these recurring problems that cause people to end up with broken environments.

Perhaps `debuginfo()` could be better.

Reply:
Yeah, I think `debuginfo` is a nicer name for this use case. :smile: Please put it in a package extension (`RCallInteractiveUtilsExt`) and ping me for re-review.

You'll have to add `InteractiveUtils` to the test environment for testing.

Reply:
I tried out what you suggested with the package extension. It seems to only work after explicitly running `using InteractiveUtils` within the REPL which seems undesirable.

I'm not totally sure with what the problem with using `Main` from a module is. Is there some information somewhere telling us that this is not allowed? Does it not work for you? I've found that for developer tools that are intended to be installed into people's global environment, it's fine to grab them from `Main`. I've seen this pattern about the place too. Maybe this is an exception to the rule?

I've updated it with a `isdefined(...)` check which should help to prevent incorrect usage in the common case.